OptimizelySDK.Tests/NotificationTests/NotificationCenterTests.cs

```diff
@@ -0,0 +1,195 @@
+
+            // Verify that notifications of other types will also gets added successfully.
+            NotificationCenter.AddNotification(notificationTypeTrack, TestNotificationCallbacks.TestTrackCallback);
+            Assert.AreEqual(3, NotificationCenter.NotificationsCount);
+        }
+
+        [Test]
+        public void TestAddSameNotificationListenerMultipleTimes()
+        {
+            var notificationTypeDecision = NotificationCenter.NotificationType.Decision;
+            NotificationCenter.AddNotification(notificationTypeDecision, TestNotificationCallbacks.TestDecisionCallback);
+
+            // Verify that adding same callback multiple times will gets failed.
+            Assert.AreEqual(-1, NotificationCenter.AddNotification(notificationTypeDecision,
+                TestNotificationCallbacks.TestDecisionCallback));
+            Assert.AreEqual(1, NotificationCenter.NotificationsCount);
+        }
+
+        [Test]
+        public void TestClearNotifications()
+        {
+            var notificationTypeDecision = NotificationCenter.NotificationType.Decision;
+            var notificationTypeTrack = NotificationCenter.NotificationType.Track;
+
+            // Add decision notifications.
+            NotificationCenter.AddNotification(notificationTypeDecision, TestNotificationCallbacks.TestDecisionCallback);
+            NotificationCenter.AddNotification(notificationTypeDecision, TestNotificationCallbacks.TestAnotherDecisionCallback);
+
+            // Add track notification.
+            NotificationCenter.AddNotification(notificationTypeTrack, TestNotificationCallbacks.TestTrackCallback);
+
+            // Verify that callbacks added successfully.
+            Assert.AreEqual(3, NotificationCenter.NotificationsCount);
+
+            // Verify that only decision callbacks are removed.
+            NotificationCenter.ClearNotifications(notificationTypeDecision);
+            Assert.AreEqual(1, NotificationCenter.NotificationsCount);
+        }
+
+        [Test]
+        public void TestClearAllNotifications()
+        {
+            var notificationTypeDecision = NotificationCenter.NotificationType.Decision;
+            var notificationTypeTrack = NotificationCenter.NotificationType.Track;
+
+            // Add decision notifications.
+            NotificationCenter.AddNotification(notificationTypeDecision, TestNotificationCallbacks.TestDecisionCallback);
+            NotificationCenter.AddNotification(notificationTypeDecision, TestNotificationCallbacks.TestAnotherDecisionCallback);
+
+            // Add track notification.
+            NotificationCenter.AddNotification(notificationTypeTrack, TestNotificationCallbacks.TestTrackCallback);
+
+            // Verify that callbacks added successfully.
+            Assert.AreEqual(3, NotificationCenter.NotificationsCount);
+
+            // Verify that ClearAllNotifications remove all the callbacks.
+            NotificationCenter.ClearAllNotifications();
```

Comment:
call clearNotifications(notificationType) one time. and then clearAllNotifications again then.
@oakbani FYI.

Reply:
Done